packages/block-editor/src/store/actions.js

```diff
@@ -637,7 +637,12 @@ export function replaceInnerBlocks( rootClientId, blocks, updateSelection = true
  *
  * @return {Object} Action object.
  */
-export function toggleBlockMode( clientId ) {
+export function * toggleBlockMode( clientId ) {
+	const canToggleBlockMode = yield select( 'core/block-editor', 'canToggleBlockMode', clientId );
```

Comment:
@talldan moved the check to action so it's not repeated.

Reply:
This looks good to me, what about the code editing mode check?